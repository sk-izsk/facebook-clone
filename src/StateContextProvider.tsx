import React, { createContext, useContext, useReducer } from 'react';

export interface StateContextProviderProps {
  initialState?: any;
  reducer?: any;
}

const StateContext = createContext(undefined);

const StateContextProvider: React.FC<StateContextProviderProps> = ({ initialState, reducer, children }) => {
  return <StateContext.Provider value={useReducer(reducer, initialState) as any}>{children}</StateContext.Provider>;
};

const useStateValue: any = () => useContext(StateContext);

export { StateContextProvider, StateContext, useStateValue };
