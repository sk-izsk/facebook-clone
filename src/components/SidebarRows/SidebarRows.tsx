import { Avatar, Box, makeStyles, Typography } from '@material-ui/core';
import React from 'react';
import { CustomTheme } from '../../theme/muiTheme';

export interface SidebarRowProps {
  src?: string;
  Icon?: JSX.Element;
  title?: string;
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  sidebarRow: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(1.2),
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: theme.palette.grey[100],
      borderRadius: 10,
    },
  },
  title: {
    fontWeight: 600,
    marginLeft: theme.spacing(2.5),
    fontSize: 15,
  },
}));

const SidebarRow: React.FC<SidebarRowProps> = ({ Icon, src, title }) => {
  const classes = useStyles();
  return (
    <Box className={classes.sidebarRow}>
      {src && <Avatar src={src} />}
      {Icon && Icon}
      <Typography className={classes.title} variant='h6'>
        {title}
      </Typography>
    </Box>
  );
};

export { SidebarRow };
