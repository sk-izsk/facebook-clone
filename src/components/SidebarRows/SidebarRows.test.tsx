import { render } from '@testing-library/react';
import React from 'react';
import { StateContextProvider } from '../../StateContextProvider';
import { SidebarRow } from './SidebarRows';

const mockUser = {
  image: 'fake image url',
  message: 'fake message',
  profilePic: 'fake profile pic',
  timestamp: 'fake time stamp',
  username: 'fake username',
};

test('SidebarRow renders correctly', () => {
  const { asFragment } = render(
    <StateContextProvider initialState={{ user: mockUser }}>
      <SidebarRow />
    </StateContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
