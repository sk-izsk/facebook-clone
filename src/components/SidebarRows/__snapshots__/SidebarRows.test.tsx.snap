// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarRow renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-sidebarRow-1"
  >
    <h6
      class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
    />
  </div>
</DocumentFragment>
`;
